weblate/templates/widgets.html

```diff
@@ -23,110 +24,76 @@ <h2>{% trans "Promoting translation projects" %}</h2>
-    :alt: {% trans "Translation status" %}
-    :target: {{ engage_url }}</textarea>
-              {% endfor %}
-            </div>
-
-            <div class="tab-pane" id="{{ widget.name }}-textile-code">
-              {% for color in widget.colors %}
-                <textarea class="code-example form-control">
-!{{ color.url }}!:{{ engage_url }}</textarea>
-              {% endfor %}
-            </div>
-
-            <div class="tab-pane" id="{{ widget.name }}-url-code">
-              {% for color in widget.colors %}
-                <textarea class="code-example form-control">
-{{ color.url }}</textarea>
-              {% endfor %}
-            </div>
-
-          </div>
-
+        </select>
+        <label for="translation_language" class="form-label mt-2">{% trans "Language" %}</label>
+        <select id="translation_language" class="select-tab form-control">
+          {% for language in form.lang.field.choices %}
+            <option value="{{ language.0 }}">{% trans language.1 %}</option>
+          {% endfor %}
+        </select>
+        <label for="component" class="form-label mt-2">{% trans "Component" %}</label>
+        <select id="component" class="select-tab form-control">
+          {% for component in form.component.field.choices %}
+            <option value="{{ component.0 }}">{% trans component.1 %}</option>
+          {% endfor %}
+        </select>
+        <label for="code_language" class="form-label mt-2">{% trans "Code" %}</label>
+        <select id="code_language" class="select-tab form-control">
+          <option value="html">{% trans "HTML" %}</option>
+          <option value="bb-code">{% trans "BBCode" %}</option>
+          <option value="mdk">{% trans "Markdown" %}</option>
+          <option value="rst">{% trans "reStructuredText" %}</option>
+          <option value="textile-code">{% trans "Textile code" %}</option>
+          <option value="url">{% trans "Image URL" %}</option>
+        </select>
+        <label for="color_select" class="form-label mt-2">{% trans "Color" %}</label>
+        <select id="color_select" class="form-control">
+          <!-- Options will be populated dynamically -->
+        </select>
+        <div id="extra-parameters"></div>
+      </div>
+    </div>
+
+    <div class="col-md-8">
+      <div class="widget-card bg-white p-3">
+        <h4>Live Preview</h4>
+        <div class="preview-box" id="preview">
+          <img id="widgetImage"
+               src="{{ image_src }}"
+               alt="{% trans "Translation status" %}"
+               style="max-width: 500px" />
         </div>
+      </div>
 
+      <div class="widget-card bg-white p-3">
+        <h4>Embed Code</h4>
+        <textarea id="embedCode" class="code-example form-control" rows="3" readonly>
+        </textarea>
       </div>
-    {% endfor %}
+    </div>
   </div>
 
 {% endblock %}
+
+{% block extra_script %}
+  <script type="application/json" id="widgets-data">{{ widgets_json|safe }}</script>
```

Comment:
I don't think safe should be used here. The JSON can still contain markup and that would break. Also passing this as data attribute would be better approach as that would not need inline scripts in CSP and you would have no issues with escaping.